src/meta/Dimensions.ts

```diff
@@ -0,0 +1,63 @@
+import { Base } from './Base';
+
+export interface DimensionResults {
```

Comment:
should we perhaps make these slightly nicer than the dom ones in terms of accessing? ie have separate objects for each, like `size`, `position`, `scroll` etc?